docs/02.API-REFERENCE.md

```diff
@@ -696,29 +696,51 @@ typedef jerry_value_t (*jerry_external_handler_t) (const jerry_call_info_t *call
 
 - [jerry_create_external_function](#jerry_create_external_function)
 
+## jerry_value_free_callback_t
+
+**Summary**
+
+Native free callback of generic value types.
+
+*Note*:
+ - Referred values by this method must have at least 1 reference. (Correct API usage satisfies this condition)
+
+**Prototype**
+
+```c
+typedef void (*jerry_value_free_callback_t) (void *native_p);
+```
+
+*New in version [[NEXT_RELEASE]]*.
+
+**See also**
+
+- [jerry_create_external_string](#jerry_create_external_string)
+- [jerry_create_external_string_sz](#jerry_create_external_string_sz)
+- [jerry_create_arraybuffer_external](#jerry_create_arraybuffer_external)
+
 ## jerry_object_native_free_callback_t
 
 **Summary**
 
-Native free callback of an object. It is used in `jerry_object_native_info_t` and for external Array buffers.
+Native free callback of an object. It is part of the type information provided by `jerry_object_native_info_t`.
 
 *Note*:
  - Referred values by this method must have at least 1 reference. (Correct API usage satisfies this condition)
 
 **Prototype**
 
 ```c
-typedef void (*jerry_object_native_free_callback_t) (void *native_p);
+typedef void (*jerry_object_native_free_callback_t) (void *native_p, struct jerry_object_native_info_t *info_p);
```

Comment:
We should document the arguments just like it is done for the `jerry_external_handler_t` type.